src/main/java/org/jenkinsci/plugins/githubautostatus/model/BuildStage.java

```diff
@@ -21,40 +21,60 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.jenkinsci.plugins.githubautostatus;
+package org.jenkinsci.plugins.githubautostatus.model;
 
+import com.google.gson.annotations.SerializedName;
 import hudson.model.Run;
+import org.eclipse.jgit.annotations.NonNull;
+import org.jenkinsci.plugins.githubautostatus.notifiers.BuildNotifierConstants;
+
 import java.util.HashMap;
 import java.util.Map;
-import org.jenkinsci.plugins.githubautostatus.notifiers.BuildState;
 
 /**
  *
  * @author jxpearce
  */
-public class BuildStageModel {
-    
+public class BuildStage {
```

Comment:
I'm curious about the reason for the rename. I'm not particularly tied to any name, but doing this change in conjunction with the added provider makes the PR a little more difficult to review. Not a huge deal, but something to keep in mind for the future